docs/tutorials/using_macros.md

```diff
@@ -0,0 +1,57 @@
+
+## Reading Values
+Macros can be read like any other python value. Both `gm` and `macros` are [Addict](https://github.com/mewwts/addict) dictionary-like objects, and whose keyword-indexed values can be queried via dot notation. Note that for `macros`, this is a nested addict object and represents values as keyword-entries indexed by the relative path of the module it is specified in, expressed in dot notation. For example, to read a macro from `omnigibson/prims/entity_prim.py`, you would query `macros.prims.entity_prim.<MACRO_NAME>`. We provide an example snippet of code below:
+
+??? code "read_macros.py"
+    ``` python linenums="1"
+
+    from omnigibson.macros import gm, macros
+
+    # Read some global macros
+    print(f"Is headless: {gm.HEADLESS}")
+    print(f"Physics freq: {gm.DEFAULT_PHYSICS_FREQ}")
+
+    # Read some module-level macros
+    print(f"Object sleep threshold: {macros.prims.entity_prim.DEFAULT_SLEEP_THRESHOLD}")
+    print(f"Visual particle removal limit: {macros.object_states.particle_modifier.VISUAL_PARTICLES_REMOVAL_LIMIT}")
+
+    # Print out all global macros
+    print(f"All global macros:\n\n{gm}")
+
+    # Print out all module-level macros
+    print(f"All module-level macros:\n\n{macros}")
+    ```
+
+## Setting Values
+Macros can be set like any other python value, and utilize the same convention for accessing as stated above. However, a key caveat is that once a macro is read, it **cannot** be set. This is to guarantee consistent runtime performance and avoid silent bugs due to macros being updated but not read by relevant downstream use cases. For example, `gm.HEADLESS` is read when `og.launch()` is called to start up the **OmniGibson** application, and cannot be modified afterward. If there was no guard in place, setting this macro after **OmniGibson** is launched would result in an inconsistency between the observed state (i.e.: the mode requested when **OmniGibson** was originally launched) and the read state (modified value of `gm.HEADLESS`). In general, we recommend setting all macros *before* `og.launch()` is called. We provide an example snippet of code below:
+
+??? code "set_macros.py"
+    ``` python linenums="1"
+
+    import omnigibson as og
+    from omnigibson.macros import gm, macros
+
+    # Set some global macros
+    gm.HEADLESS = True
+    gm.DEFAULT_PHYSICS_FREQ = 60
+
+    # Set some module-level macros
+    macros.prims.entity_prim.DEFAULT_SLEEP_THRESHOLD = 0.002
+    macros.object_states.particle_modifier.VISUAL_PARTICLES_REMOVAL_LIMIT = 80
+
+    # Now launch omnigibson
+    og.launch()
+    ```
```

Comment:
maybe we should also add
```
env = og.Environment({"scene": {"type": "Scene"}})
```

I felt like our users will very rarely actually call `og.launch()`.

Reply:
ooo good point, updated!